lib/presentation/feed/widgets/feed_card.dart

```diff
@@ -0,0 +1,248 @@
+                        backgroundImage: NetworkImage(feed.picture!),
+                      ),
+              ),
+            ),
+            Expanded(
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: <Widget>[
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: <Widget>[
+                      Text(
+                        '${feed.fullname!.trim()} solved',
+                        style: Theme.of(context).textTheme.headline6,
+                      ),
+                      Text(
+                        DateFormat('dd-MM-yyyy kk:mm').format(
+                          feed.submissions![0].createdAt!,
+                        ),
+                        style: Theme.of(context).textTheme.headline6,
+                      ),
+                    ],
+                  ),
+                  Padding(
+                    padding: EdgeInsets.symmetric(vertical: 5.h),
+                    child: Text(
+                      '${feed.name}',
+                      style: Theme.of(context)
+                          .textTheme
+                          .headline4
+                          ?.copyWith(fontSize: 16.sp),
+                      overflow: TextOverflow.ellipsis,
+                      maxLines: 2,
+                    ),
+                  ),
+                  Row(
+                    children: <Widget>[
+                      Text(
+                        'on ',
+                        style: Theme.of(context)
+                            .textTheme
+                            .headline6
+                            ?.copyWith(fontSize: 12.sp),
+                      ),
+                      CircleAvatar(
+                        backgroundColor: Colors.transparent,
+                        backgroundImage: AssetImage(
+                          ContestUtil.getPlatformIcon(
+                            ContestUtil.getPlatformNamefromUrl(
+                              feed.url,
+                            ),
+                          ),
+                        ),
+                        radius: 10.r,
+                      ),
+                      SizedBox(
+                        width: 150.w,
+                        child: Text(
+                          _makeNonBreaking(
+                            ' ${ContestUtil.getPlatformName(ContestUtil.getPlatformNamefromUrl(feed.url))} |'
+                            ' ${feed.language}',
+                          ),
+                          style: Theme.of(context)
+                              .textTheme
+                              .headline6
+                              ?.copyWith(fontSize: 12.sp),
+                          overflow: TextOverflow.ellipsis,
+                        ),
+                      ),
+                      const Spacer(),
+                      ValueListenableBuilder<bool>(
```

Comment:
What's its purpose here?

Reply:
This widget will build the icon according to the current state of the ExpansionTile.